src/ipc/ipc4/dai.c

```diff
@@ -367,6 +367,13 @@ int dai_position(struct comp_dev *dev, struct sof_ipc_stream_posn *posn)
 	return 0;
 }
 
+int dai_position(struct comp_dev *dev, struct sof_ipc_stream_posn *posn)
+{
+	struct dai_data *dd = comp_get_drvdata(dev);
+
+	return dai_zephyr_position(dd, dev, posn);
+}
+
```

Comment:
I think we can probably make some of these light function static inline in the header.

Reply:
agree, just because this is an formal inerface for driver, if move ipc4 to inline, then legacy part also need inline, it is not a performance function, so overhead is not a problem.